plugin/pkg/admission/broker/authsarcheck/admission_test.go

```diff
@@ -244,4 +244,178 @@ func TestAdmissionBroker(t *testing.T) {
 			t.Errorf("Create test '%s' reports: Unexpected error returned from admission handler: %v", tc.name, err)
 		}
 	}
+
+	// Anonymous struct fields:
+	// name: short description of the testing
+	// broker: a fake broker object
+	// allowed: flag for whether or not the broker should be admitted
+	namespacedCases := []struct {
+		name     string
+		broker   *servicecatalog.ServiceBroker
+		userInfo *user.DefaultInfo
+		allowed  bool
+	}{
+		{
+			name: "namespace broker with no auth",
+			broker: &servicecatalog.ServiceBroker{
+				ObjectMeta: metav1.ObjectMeta{
+					Namespace: "test-ns",
+					Name:      "test-broker",
+				},
+				Spec: servicecatalog.ServiceBrokerSpec{
+					CommonServiceBrokerSpec: servicecatalog.CommonServiceBrokerSpec{
+						URL:            "http://example.com",
+						RelistBehavior: "Manual",
+					},
+				},
+			},
+			userInfo: &user.DefaultInfo{
+				Name:   "system:serviceaccount:test-ns:catalog",
+				Groups: []string{"system:serviceaccount", "system:serviceaccounts:test-ns"},
+			},
+			allowed: true,
+		},
+		{
+			name: "namespace broker with basic auth, user authenticated",
+			broker: &servicecatalog.ServiceBroker{
+				ObjectMeta: metav1.ObjectMeta{
+					Namespace: "test-ns",
+					Name:      "test-broker",
+				},
+				Spec: servicecatalog.ServiceBrokerSpec{
+					AuthInfo: &servicecatalog.ServiceBrokerAuthInfo{
+						Basic: &servicecatalog.BasicAuthConfig{
+							SecretRef: &servicecatalog.LocalObjectReference{
+								Name: "test-secret",
+							},
+						},
+					},
+					CommonServiceBrokerSpec: servicecatalog.CommonServiceBrokerSpec{
+						URL:            "http://example.com",
+						RelistBehavior: "Manual",
+					},
+				},
+			},
+			userInfo: &user.DefaultInfo{
+				Name:   "system:serviceaccount:test-ns:catalog",
+				Groups: []string{"system:serviceaccount", "system:serviceaccounts:test-ns"},
+			},
+			allowed: true,
+		},
+		{
+			name: "namespace broker with bearer token, user authenticated",
+			broker: &servicecatalog.ServiceBroker{
+				ObjectMeta: metav1.ObjectMeta{
+					Namespace: "test-ns",
+					Name:      "test-broker",
+				},
+				Spec: servicecatalog.ServiceBrokerSpec{
+					AuthInfo: &servicecatalog.ServiceBrokerAuthInfo{
+						Bearer: &servicecatalog.BearerTokenAuthConfig{
+							SecretRef: &servicecatalog.LocalObjectReference{
+								Name: "test-secret",
+							},
+						},
+					},
+					CommonServiceBrokerSpec: servicecatalog.CommonServiceBrokerSpec{
+						URL:            "http://example.com",
+						RelistBehavior: "Manual",
+					},
+				},
+			},
+			userInfo: &user.DefaultInfo{
+				Name:   "system:serviceaccount:test-ns:catalog",
+				Groups: []string{"system:serviceaccount", "system:serviceaccounts:test-ns"},
+			},
+			allowed: true,
+		},
+		{
+			name: "namespace broker with bearer token, unauthenticated user",
```

Comment:
what's the significants of bearer token and unauthenticated together? Is there an unauthenticated on it's own, I think it's just copy and paste from above and the cases don't necessarily make sense.

Reply:
the bearer token is platform <-> broker, not user <-> platform